apps/desktop-wallet/src/api/queries/tokenQueries.ts

```diff
@@ -78,29 +83,37 @@ export const combineTokenTypeQueryResults = (results: UseQueryResult<TokenInfo |
   ...combineIsLoading(results)
 })
 
-export const fungibleTokenMetadataQuery = ({ id, skip }: TokenQueryProps) =>
+export const fungibleTokenMetadataQuery = ({ id, networkId, skip }: TokenQueryProps) =>
   queryOptions({
     queryKey: ['token', 'fungible', 'metadata', id],
+    staleTime: Infinity,
+    // We don't want to delete the fungible token metadata when the user navigates away components that need them.
+    gcTime: Infinity,
+    meta: { isMainnet: networkId === 0 },
     queryFn: !skip
       ? async () => {
           const tokenMetadata = await batchers.ftMetadataBatcher.fetch(id)
 
           return tokenMetadata ? convertTokenDecimalsToNumber(tokenMetadata) : null
         }
-      : skipToken,
-    staleTime: Infinity
+      : skipToken
   })
 
-export const nftMetadataQuery = ({ id, skip }: TokenQueryProps) =>
+export const nftMetadataQuery = ({ id, networkId, skip }: TokenQueryProps) =>
   queryOptions({
     queryKey: ['token', 'non-fungible', 'metadata', id],
-    queryFn: !skip ? async () => (await batchers.nftMetadataBatcher.fetch(id)) ?? null : skipToken,
-    staleTime: Infinity
+    staleTime: Infinity,
+    gcTime: Infinity, // We don't want to delete the NFT metadata when the user navigates away from NFT components.
+    meta: { isMainnet: networkId === 0 },
```

Comment:
I'm not a huge fan of this. It's not super DRY. And what if we forget it?

Reply:
How do you recommend we do it?

Reply:
If we forget it, the data will be persisted. That's the worst case

Reply:
I wanted to use the `networkId` from the `queryKey`, but each query can have it in a different index in the array... I thought a more standardized way would be to use `meta`.

Reply:
Would you prefer a function

```ts
export const generateNetworkQueryMeta = (networkId?: number) => ({
  isMainnet: networkId === 0
})
```

and then using it like so:

```ts
const { data, isLoading } = useQuery({
    queryKey: ['tokenList', { networkId }],
    staleTime: ONE_DAY_MS,
    gcTime: Infinity,
    meta: generateNetworkQueryMeta(networkId),
```